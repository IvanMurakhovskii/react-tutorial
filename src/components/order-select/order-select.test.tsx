import React from 'react';
import { shallow } from "enzyme";
import OrderSelect from './order-select';
import OrderEnum from '@/emums/order-enum';

const mockMyEventHandler = jest.fn()

const item = <OrderSelect onOrderChange={mockMyEventHandler} order={OrderEnum.ASC} />

const wrapper = shallow(item);

describe("OrderSelect", () => {
    it("render item with data", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("simulate dropdown list change", () => {
        wrapper.find('.order-select').simulate('change', { target: { value: 'ASC' } });
        expect(mockMyEventHandler).toHaveBeenCalledWith('ASC');
    });

});