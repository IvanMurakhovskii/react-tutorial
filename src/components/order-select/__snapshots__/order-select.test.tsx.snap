// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSelect render item with data 1`] = `
<Styled(div)>
  <WithStyles(ForwardRef(FormControl))
    size="small"
    variant="standard"
  >
    <WithStyles(ForwardRef(InputLabel))
      id="labelId"
    >
      Сортировка
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(Select))
      className="order-select"
      labelId="labelId"
      onChange={[Function]}
      value="ASC"
    >
      <WithStyles(ForwardRef(MenuItem))
        key="1"
        value="ASC"
      >
        По возрастанию
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        key="2"
        value="DESC"
      >
        По убыванию
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        key="3"
        value="IMPORTANT"
      >
        Важные
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
  </WithStyles(ForwardRef(FormControl))>
</Styled(div)>
`;
